java/AxivityReader.java

```diff
@@ -330,18 +348,37 @@ private static LocalDateTime getCwaTimestamp(int cwaTimestamp, int fractional, i
         int mins = (int) ((cwaTimestamp >> 6) & 0x3f);
         int secs = (int) ((cwaTimestamp) & 0x3f);
         tStamp = LocalDateTime.of(year, month, day, hours, mins, secs);
-        tStamp = tStamp.plusMinutes(timeZoneOffset);
         // add 1/65536th fractions of a second
         tStamp = tStamp.plusNanos(secs2Nanos(fractional / 65536.0));
         return tStamp;
     }
 
 
-    private static LocalDateTime cwaHeaderLoggingStartTime(ByteBuffer buf, int timeZoneOffset) {
+    private static LocalDateTime cwaHeaderLoggingStartTime(ByteBuffer buf) {
         long delayedLoggingStartTime = getUnsignedInt(buf, 13);
-        return getCwaTimestamp((int) delayedLoggingStartTime, 0, timeZoneOffset);
+        return getCwaTimestamp((int) delayedLoggingStartTime, 0);
+    }
+
+
+    private static void setTimeZone(String timeZone) {
+        zoneId = ZoneId.of(timeZone);
+        rules = zoneId.getRules();
     }
 
 
+    private static void setSessionStart(LocalDateTime ldt) {
+        sessionStart = ldt.atZone(zoneId);
+        sessionStartDST = rules.isDaylightSavings(sessionStart.toInstant());
+    }
+
+
+    private static ZonedDateTime zonedWithDSTCorrection(LocalDateTime ldt) {
```

Comment:
@chanshing - can you move this to the abstract DeviceReader.java class?
i.e. we would like this to be available to all of AxivityReader.java, ActigraphReader.java, etc.